pennylane/transforms/measurements_from_computational_basis.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Transform to diagonalize measurements on a tape and transform the measurements
+to come from readout in the computational basis returned as samples or counts."""
+
+import pennylane as qml
+from pennylane.transforms import transform
+
+
+@transform
+def measurements_from_samples(tape):
+    r"""Replace all measurements from a tape with a single sample measurement,
+    and adds postprocessing functions for each original measurement.
+
+    .. note::
+        This transform diagonalizes all the operations on the tape. An error will
+        be raised if non-commuting terms are encountered. To avoid non-commuting
+        terms in circuit measurements, the :func:`split_non_commuting <pennylane.transforms.split_non_commuting>`
+        transform can be applied.
+
+    Args:
+        tape (QNode or QuantumTape or Callable): A quantum circuit.
+
+    Returns:
+        qnode (QNode) or quantum function (Callable) or tuple[List[QuantumTape], function]: The
+        transformed circuit as described in :func:`qml.transform <pennylane.transform>`.
+
+    **Example**
```

Comment:
Can we change the example here to be more user-oriented? E.g.,

```python
@qml.qnode(qml.device("default.qubit", wires=1, shots=100))
def circuit():
    qml.H(wires=0)
    return qml.probs(wires=0)
```

```
>>> print(qml.draw(circuit)())
0: ──H─┤  Probs
>>> circuit_samples = qml.transforms.measurements_from_samples(circuit)
>>> print(circuit_samples())
[0.54 0.46]
>>> print(qml.draw(circuit_samples)())
0: ──H─┤  Sample
```

Reply:
Also wouldn't hurt to have an example (in usage details) about what to do in the scenario mentioned in the note 👀 

![Screenshot 2025-04-22 at 1 25 38 PM](https://github.com/user-attachments/assets/adc7c4e3-4545-4179-8e0c-82a432802d4a)

Reply:
Same comment for the `measurements_from_counts` transform 😁